presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -717,9 +717,12 @@ private PlanWithProperties planPartitionedJoin(JoinNode node, List<Symbol> leftS
                 Partitioning rightPartitioning = left.getProperties().translate(createTranslator(leftToRight)).getNodePartitioning().get();
                 right = node.getRight().accept(this, PreferredProperties.partitioned(rightPartitioning));
                 if (!right.getProperties().isCompatibleTablePartitioningWith(left.getProperties(), rightToLeft::get, metadata, session)) {
-                    // TODO: support materialization when one of the tables is pre-bucketed
                     right = withDerivedProperties(
-                            partitionedExchange(idAllocator.getNextId(), REMOTE_STREAMING, right.getNode(), new PartitioningScheme(rightPartitioning, right.getNode().getOutputSymbols())),
+                            partitionedExchange(
+                                    idAllocator.getNextId(),
+                                    selectExchangeScopeForPartitionedRemoteExchange(right.getNode()),
```

Comment:
We should have a session property that allows to force the selected materialized partitioning. We can have it as a separate PR though.

Reply:
@arhimondr : Yup. This about `join_partitoning_selection_strategy`:
- LEFT_RIGHT_SPECIFIED
- ALWAYS_SPECIFIED

in the future, we might want to have `MIN` / `MAX`....